### ISSUE TITLE
`HashFunction::complete_with_mask` overwrites previous operators

### ISSUE THREAD

**Issue by jdreo** [yoann-dufresne/rHashGen#6]
As of acaeed9,  `HashFunction::complete_with_mask`  skips existing `Masking` operators.

The name implies that the function adds additional operators, but this cannot be true if the hasf function has two `Masking`.

For instance, the following test is failing:
```cpp
TEST(HashingTest, HandPicked)
{
    using myuint = uint32_t;
    HashFunction<myuint> hf(31);
        hf.add_operator(std::make_unique<Masking      <myuint>>(26));
        hf.add_operator(std::make_unique<Masking      <myuint>>( 2));
        hf.add_operator(std::make_unique<AddShift     <myuint>>(22, 31));

    const size_t initial_size = hf.size();
    hf.complete_with_masks();
    const size_t completed_size = hf.size();
    EXPECT_GE(completed_size, initial_size); // <- Fails here. But is it always true?
    HashFunction<myuint> rhf{hf.invert()};
    EXPECT_GT(rhf.size(), 0);

    // Set up the random number generator
    std::random_device rd;  // Obtain a random number from hardware
    auto const seed{rd()};
    std::mt19937 gen(seed); // Seed the generator
    std::uniform_int_distribution<myuint> value_dist(0, (1U << 31) - 1);

    for (size_t i{0} ; i<100 ; i++)
    {
        myuint value {static_cast<myuint>(value_dist(gen))};
        myuint hashed {hf.apply(value)};
        myuint recovered {rhf.apply(hashed)};
        EXPECT_EQ(value, recovered)  << "Failed with random seed " << seed;
    }
}
```

Is it a bug or a feature? If it's a feature, it should be documented (and maybe the name of the function changed?).

**Comment by jdreo** [comment]
Another problem occurs when one (like a solver ;) picks a hash function consisting only of masks, since the reverse hash is empty. Is it expected?

**Comment by jdreo** [comment]
This is because Masking should not be part of the search domain. Fixed by #9
